src/utils/DateUtils.ts

```diff
@@ -29,138 +155,231 @@ class DefaultDateToStringOptions extends Options implements IDateToStringOptions
+      return moment(Number(date)).toDate();
     } else if (_.isString(date)) {
-      return new Date(<string>date.replace('@', ' '));
-    } else {
-      return undefined;
+      const formats = ['YYYY/MM/DD@HH:mm:ssZ', moment.ISO_8601];
+      const dateMoment = moment(date, formats);
+      return dateMoment.toDate();
     }
   }
 
-  static dateForQuery(date: Date): string {
-    return date.getFullYear() + '/' + DateUtils.padNumber((date.getMonth() + 1).toString()) + '/' + DateUtils.padNumber(date.getDate().toString());
-  }
+  public static setLocale(): void {
+    let currentLocale = String['locale'];
 
-  private static padNumber(num: string, length: number = 2): string {
-    while (num.length < length) {
-      num = '0' + num;
+    // Our cultures.js directory contains 'no' which is the equivalent to 'nn' for momentJS
+    if (currentLocale.toLowerCase() == ('no')) {
+      currentLocale = 'nn';
+    } else if (currentLocale.toLowerCase() == 'es-es') {
+      // Our cultures.js directory contains 'es-es' which is the equivalent to 'es' for momentJS
+      currentLocale = 'es';
     }
-    return num;
+
+    moment.locale(currentLocale);
+
+    moment.updateLocale(currentLocale, {
+      calendar: {
+        lastDay: `[${l('Yesterday')}]`,
+        sameDay: `[${l('Today')}]`,
+        nextDay: `[${l('Tomorrow')}]`
+      }
+    });
+  }
+
+  /**
+   * Receives a Date and formats into the standard string required for queries.
+   * @param date The date to convert into a string.
+   * @returns {string} Corresponds to the Date in a `YYYY/MM/DD` format.
+   */
+  static dateForQuery(date: Date): string {
+    DateUtils.setLocale();
+    const dateMoment = moment(date).format('YYYY/MM/DD');
+    return dateMoment;
   }
 
+  /**
+   * Keeps only the date, removing the time information.
+   * @param date The date to crop.
+   * @returns {Date} The cropped date, without the time information.
+   */
   static keepOnlyDatePart(date: Date): Date {
-    return new Date(date.getFullYear(), date.getMonth(), date.getDate());
+    DateUtils.setLocale();
+    const dateMoment = moment(date);
+    return new Date(dateMoment.year(), dateMoment.month(), dateMoment.date());
   }
 
+  /**
+   * Adds offset to a Date, counted in days.
+   * @param date The date to offset.
+   * @param offset The amount of days to add or remove to the date.
+   * @returns {Date} The modified date.
+   */
   static offsetDateByDays(date: Date, offset: number): Date {
-    var newDate = new Date(date.valueOf());
-    newDate.setDate(newDate.getDate() + offset);
+    return moment(date).add(offset, 'days').toDate();
+  }
 
-    return newDate;
+  private static isTodayYesterdayOrTomorrow(d: Date, options?: IDateToStringOptions): boolean {
+    const dateOnly = moment(DateUtils.keepOnlyDatePart(d));
+    const today = moment(DateUtils.keepOnlyDatePart(options.now));
+    return dateOnly.diff(today, 'days') == 0 || dateOnly.diff(moment(today), 'days') == 1 ||
```

Comment:
could have calculated the diff only once, and then compare with that.

so
        const daysDifference = dateOnly.diff(today, 'days'); 
        return daysDifference == 0 || daysDifference  == 1 || daysDifference  == -1